x-pack/plugins/rule_registry/server/utils/create_lifecycle_executor.ts

```diff
@@ -230,22 +233,30 @@ export const createLifecycleExecutor =
       alertIds.map((alertId) => {
         const alertData = trackedAlertsDataMap[alertId];
         const currentAlertData = currentAlerts[alertId];
+        const trackedAlert = state.trackedAlerts[alertId];
 
         if (!alertData) {
           logger.debug(`[Rule Registry] Could not find alert data for ${alertId}`);
         }
 
-        const isNew = !state.trackedAlerts[alertId];
-        const isRecovered = !currentAlerts[alertId];
+        const isNew = !trackedAlert;
+        const isRecovered = !currentAlertData;
         const isActive = !isRecovered;
 
-        const { alertUuid, started } = !isNew
+        const { alertUuid: alertUuidCalc, started } = !isNew
           ? state.trackedAlerts[alertId]
           : {
-              alertUuid: newAlertUuids[alertId] || v4(),
+              alertUuid: alertUuidMap.get(alertId),
               started: commonRuleFields[TIMESTAMP],
             };
 
+        const alertUuid = alertUuidCalc ?? v4();
```

Comment:
This shouldn't happen, but if for some reason the alert uuid calculation does not work, revert to having the rule registry generate it and log a message that it did.